c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -2151,6 +2149,46 @@ class DeviceCachingAllocator {
     return blocks;
   }
 
+  void maybe_make_new_pool_and_inc_use_count(MempoolId_t mempool_id) {
+    auto it = graph_pools.find(mempool_id);
+    if (it == graph_pools.end()) {
+      // mempool_id does not reference an existing pool.
+      // Make a new pool for CUDAGraph capture or torch.cuda.use_mem_pool
+      // usage. use_count is initially 0. Increment it to establish the usage.
+      auto inserted =
+          graph_pools.emplace(mempool_id, std::make_unique<PrivatePool>());
+      TORCH_INTERNAL_ASSERT(inserted.second);
+      inserted.first->second->use_count++;
```

Comment:
This is now different

Reply:
At first, it looks like there is a net change in refcount with the change here. But actually, there isn't, because you also changed PrivatePool to now be created with zero refcount? Why not just keep PrivatePool created with refcount 1 and delete the increment here? I don't think you ever want zero refcount pools floating around?

Reply:
Yes, there isn't a net change.

> Why not just keep PrivatePool created with refcount 1 and delete the increment here?

It just looks cleaner/more explicit to me and helped in tally the pairs of incref and decref.

> I don't think you ever want zero refcount pools floating around?

Yeah, I agree. This is the only place where the pool gets created, and immediately updated to have a 1 refcount, so I thought it's ok. Zero refcount pools can't be used and will eventually hard-error here: https://github.com/pytorch/pytorch/blob/main/c10/cuda/CUDACachingAllocator.cpp#L2082.

Reply:
Changed it back to 1 and deleted the increment.